extensions/ql-vscode/src/data-extensions-editor/model-details/model-details-panel.ts

```diff
@@ -24,14 +24,21 @@ export class ModelDetailsPanel extends DisposableObject {
   }
 
   public async setState(
-    externalApiUsages: ExternalApiUsage[],
-    databaseItem: DatabaseItem,
+    externalApiUsages?: ExternalApiUsage[],
+    databaseItem?: DatabaseItem,
+    hideModeledApis?: boolean,
   ): Promise<void> {
-    await this.dataProvider.setState(externalApiUsages, databaseItem);
-    this.treeView.badge = {
-      value: externalApiUsages.length,
-      tooltip: "Number of external APIs",
-    };
+    await this.dataProvider.setState(
+      externalApiUsages,
+      databaseItem,
+      hideModeledApis,
+    );
+    if (externalApiUsages) {
```

Comment:
Hmm, I've just tried it out and realised that changing `hideModeledApis` should change the badge too, otherwise the number doesn't correctly match the data actually shown in the view.

But, we need `externalApiUsages` to be able to calculate it. This leans me towards saying we should always pass all values, i.e. instead of them all being optional, they should all be required. This will also help with the problem when switching between data extension editors. What do you think?